client/app/src/main/java/com/example/criengine/Activities/RootActivity.java

```diff
@@ -51,21 +66,25 @@ protected void onCreate(Bundle savedInstanceState) {
         // If returning from another activity, this can control which screen to navigate to.
         if (getIntent().getExtras() != null) {
             int index = ((PAGE) getIntent().getSerializableExtra("Index")).getValue();
-            viewPager.setCurrentItem(index);
+            viewPager.setCurrentItem(index, false);
         } else {
             // Returns to my books. This will be the home screen.
-            viewPager.setCurrentItem(PAGE.MY_BOOKS.getValue());
+            viewPager.setCurrentItem(PAGE.MY_BOOKS.getValue(), false);
         }
     }
 
     /**
      * FragmentStateAdapter for the ViewPager
      */
     private class RootPagerFragmentAdapter extends FragmentStateAdapter {
-        public RootPagerFragmentAdapter(@NonNull FragmentActivity fa) { super(fa); }
+        public RootPagerFragmentAdapter(@NonNull FragmentActivity fa) {
+            super(fa);
+        }
 
         @Override
-        public int getItemCount() { return 5; }
+        public int getItemCount() {
+            return 5;

```

Comment:
smells kind of magic number. any way to change this to something else?

Reply:
alrighty, I'll add it as a constant to the class

Reply:
Is that alright?

Reply:
no way to get something like PAGE.length? it's still a magic number, you just moved it out